staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/errors/statuserror.go

```diff
@@ -32,6 +33,15 @@ func ToStatusErr(webhookName string, result *metav1.Status) *apierrors.StatusErr
 		result = &metav1.Status{Status: metav1.StatusFailure}
 	}
 
+	// Make sure we don't return < 400 status codes along with a rejection
+	if result.Code < http.StatusBadRequest {
+		result.Code = http.StatusBadRequest
```

Comment:
We talked about switching this to a 500 in the api-machinery call today, but the more I consider this, I think that would be misleading to the end user. The reason their request was rejected was because their request was judged to be bad, *not* because the server had some error.

I see this as akin to what we do here:
https://github.com/kubernetes/kubernetes/blob/7289c1e97aab6a1a81f78f3b9967314625afcdc8/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go#L119-L126

if we encounter a negotiation error in the process of trying to write an error response, we don't switch to a 500, we preserve the original error code

Reply:
I personally find that argument compelling.